elfi/visualization.py

```diff
@@ -56,3 +60,129 @@ def draw_model(discrepancy_node, draw_constants=False, filename=None):
             raise ValueError('Problem with the given filename.')
 
     return dot
+
+
+def _create_axes(axes, shape, **kwargs):
+    """Checks the axes
+
+    Parameters
+    ----------
+    axes : one or an iterable of plt.Axes
+    shape : tuple of ints (x,) or (x,y)
+
+    Returns
+    -------
+    axes : np.array of plt.Axes
+    kwargs : dict
+        Input kwargs without items related to creating a figure.
+    """
+    fig_kwargs = {}
+    kwargs['figsize'] = kwargs.get('figsize', (16, 4*shape[0]))
+    for k in ['figsize', 'sharex', 'sharey', 'dpi', 'num']:
+        if k in kwargs.keys():
+            fig_kwargs[k] = kwargs.pop(k)
+
+    if axes is not None:
+        axes = np.atleast_1d(axes)
+    else:
+        fig, axes = plt.subplots(ncols=shape[1], nrows=shape[0], **fig_kwargs)
+        axes = np.atleast_1d(axes)
+    return axes, kwargs
+
+
+def _limit_params(samples, selector=None):
+    """Pick selected samples.
```

Comment:
Again the name and description feel confusing. Limiting parameters by selecting samples?